### PR TITLE
Add test for package update timeout

### PR DESCRIPTION
**Pull request by swadeley**
Closed loop test for:
Bug 1651852 - Missing timeout for "Actions::Katello::Host::Package::Update" task

### CODE REVIEW

**Comment by swadeley** [comment]
robottelo]$ pytest --show-capture=no tests/foreman/ui/test_contenthost.py -k test_actions_katello_host_package_update_timeout
====================== test session starts ================================
platform linux -- Python 3.7.6, pytest-4.6.3, py-1.8.1, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo
plugins: cov-2.8.1, services-1.3.1, mock-1.10.4, forked-1.1.3, xdist-1.31.0
collecting ... 2020-02-19 19:52:11 - conftest - DEBUG - Collected 25 test cases
2020-02-19 19:52:12 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f16633e38d0
2020-02-19 19:52:12 - robottelo.ssh - INFO - Connected to [dhcp-3-235.vms.sat.rdu2.redhat.com]
2020-02-19 19:52:12 - robottelo.ssh - INFO - >>> rpm -q satellite
2020-02-19 19:52:14 - robottelo.ssh - INFO - <<< stdout
satellite-6.7.0-5.beta.el7sat.noarch

2020-02-19 19:52:14 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f16633e38d0
2020-02-19 19:52:14 - robottelo.host_info - DEBUG - Host Satellite version: 6.7
2020-02-19 19:52:14 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 25 items / 24 deselected / 1 selected                                                                  

tests/foreman/ui/test_contenthost.py .                                                                     [100%]


============= 1 passed, 24 deselected, 2 warnings in 746.89 seconds ==================
(robottelo-master) [sjw@t480s robottelo]$

File: tests/foreman/ui/test_contenthost.py [hunk 1]
```diff
@@ -422,6 +422,44 @@ def test_positive_remove_package_group(session, vm):
...
+@tier3
+def test_actions_katello_host_package_update_timeout(session, vm):
+    """Check that Actions::Katello::Host::Package::Update task will time
+    out if goferd does not respond while attempting to update a package.
+
+    :id: 26f3ea2a-509a-4f3f-b5d7-d34b29ceb2cc
+
+    :BZ: 1651852
+
+    :expectedresults: Update task times out and error message is displayed.
```
Comment (omkarkhatavkar) [inline tests/foreman/ui/test_contenthost.py]: Just a curious question, where we are checking that the task is failing with time out? I see assumption here that we are assuming the if we see  `assert result['result'] == 'warning'` that means this is failed with time out error. 

@swadeley Do you think we need to add more assertion to ensure that is failed with timeout error only?

Comment (swadeley) [reply tests/foreman/ui/test_contenthost.py]: ACK, will investigate. Thanks for looking.

Comment (JacobCallahan) [reply tests/foreman/ui/test_contenthost.py]: @swadeley any update?

Comment (swadeley) [reply tests/foreman/ui/test_contenthost.py]: @JacobCallahan , I can get task duration from the table of tasks. I have been doing some testing with airgun, I found some widget needs updating (CSS code has changed). I might also be able to get task duration info using API, but I have been busy with other bugs. Will get back onto this in the morning.

Comment (swadeley) [reply tests/foreman/ui/test_contenthost.py]: Hello, on the task details Error tab you can see: Exception
RuntimeError: Host did not respond within 20 seconds. The task has been cancelled. Is katello-agent installed and goferd running on the Host?

But I have no idea how to get that programatically (airgun or API).

Using airgun we can read:

In [3]: session.task.read('Update package for pzyeso3djzmen9an.lab.eng.rdu2.redhat.com walrus')                   
Out[3]:
{'menu': ['', 'Monitor', 'owyzdesp', 'Default Location'],
 'flash': [],
 'current_user': 'contenthost_sDesvfJ4ux',
 'account_menu': 'contenthost_sDesvfJ4ux',
 'logout': 'Log Out',
 'breadcrumb': 'Update package for pzyeso3djzmen9an.lab.eng.rdu2.redhat.com walrus',
 'task': {'name': 'N/A',
  'result': '',
  'triggered_by': 'Ended at:',
  'execution_type': 'Immediate',
  'start_at': '1 hour ago',
  'started_at': '1 hour ago',
  'ended_at': '1 hour ago',
  'state': 'State: stopped'}}

but that info does not prove that it failed with timeout error only.

Also can read:

In [7]: session.task.read_all('table')                                                                            
Out[7]:
{'table': [{'Select all rows': 'Select row',
   'Action': 'Update package for g5vni1949pbkhpui.lab.eng.rdu2.redhat.com walrus',
   'State': 'stopped',
   'Result': 'warning',
   'Started at': 'March 25, 2020, 11:41:27 AM',
   'Duration': '21 seconds',
   'Operation': 'Cancel'}]}

Comment (swadeley) [reply tests/foreman/ui/test_contenthost.py]: I found this in the logs:

/var/log/foreman/production.log-20200326:2020-03-25T08:12:00 [E|bac|] Host did not respond within 20 seconds. The task has been cancelled. Is katello-agent installed and goferd running on the Host? (RuntimeError)

I regard that as proof. If we assert that test is present in the logs within 30 seconds of the "# Attempt to update fake package, check for warning" section, would @omkarkhatavkar approve this PR?

Comment (JacobCallahan) [reply tests/foreman/ui/test_contenthost.py]: @omkarkhatavkar is this sufficient?

**Comment by swadeley** [comment]
@omkarkhatavkar I have updated the test to include log check for error message.  Test results coming soon.

Thank you

**Comment by swadeley** [comment]
robottelo]$ git status
On branch actions_katello_host_package_update
nothing to commit, working tree clean
robottelo]$ pytest --show-capture=no tests/foreman/ui/test_contenthost.py -k test_actions_katello_host_package_update_timeout
==================== test session starts ========================================
platform linux -- Python 3.7.6, pytest-4.6.3, py-1.8.1, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo
plugins: cov-2.8.1, services-1.3.1, mock-1.10.4, forked-1.1.3, xdist-1.31.0
collecting ... 2020-04-02 09:31:29 - conftest - DEBUG - Collected 25 test cases
2020-04-02 10:31:30 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f551c59bbd0
2020-04-02 10:31:30 - robottelo.ssh - INFO - Connected to [dhcp-2-55.vms.sat.rdu2.redhat.com]
2020-04-02 10:31:30 - robottelo.ssh - INFO - >>> rpm -q satellite
2020-04-02 10:31:32 - robottelo.ssh - INFO - <<< stdout
satellite-6.7.0-7.el7sat.noarch

2020-04-02 10:31:32 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f551c59bbd0
2020-04-02 10:31:32 - robottelo.host_info - DEBUG - Host Satellite version: 6.7
2020-04-02 10:31:32 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 25 items / 24 deselected / 1 selected                                                                  

tests/foreman/ui/test_contenthost.py .                                                            [100%]


============== 1 passed, 24 deselected, 2 warnings in 745.21 seconds ==================

**Comment by swadeley** [comment]
Thank you all.

**Pull request merged**